docs/rfcs/2025-10-20-chat-relays.md

```diff
@@ -0,0 +1,256 @@
+    R ->> RSMP: 6. Create relay link
+    RSMP -->> R:
+    R ->> O: 7. Accept request<br>(x.grp.relay.acpt<br>incl. relay link)
+    O ->> OSMP: 8. Update short link<br>(add relay link)
+    OSMP -->> O:
+    O ->> R: 9. Notify relay link added<br>(x.grp.info)
+    R ->> OSMP: 10. Retrieve short link data
+    OSMP -->> R:
+    R ->> R: 11. Check relay link added
+    R ->> O: 12. Confirm to owner<br>(x.grp.relay.ready)
+end
+create participant M as Member
+O -->> M: 13. Share group short link<br>(social, out-of-band)
+
+note over O, M: New member connects
+
+M ->> OSMP: 14. Retrieve short link data
+OSMP -->> M:
+M ->> R: 15. Connect via relay link(s)
+R -->> M:
+
+note over O, M: Message forwarding
+
+O ->> R: 16. Send message
+R ->> M: 17. Forward message
+M ->> M: 18. Deduplicate message
+```
+
+Notes:
+
+- On creating group short link beforehand (step 2).
+
+  We do it for protocol simplicity - to have same logic of updating link data with each relay's link.
+
+  Alternatively owner client could create group link with first relay's link, saving a network request. This would require owner client to track state of creating the link to avoid race on receiving multiple relay links.
+
+- On adding group short link to group profile (step 3).
+
+  For protocol purposes it's only a means of informing chat relays about it (profile is passed in steps 5, x.grp.relay.inv, and 9, x.grp.info).
+
+  Alternatively it could be sent as a standalone object in initial contact request to relay (step 5, x.grp.relay.inv), or in step 9 in special event.
+
+  However, there are other arguments for having group link in profile:
+
+  - Strengthening association between link and profile. Link already contains profile in attached data, but from perspective of group profile link itself is detached. All members "see" the same link they joined via in group profile. Chat relays "see" the same link they created relay links for, and can check it for presence of their relay link at any point.
+
+  - Link is recoverable from profile, e.g. for purpose of restoring connection with group via new chat relays.
+
+  Overall it just seems a natural and convenient way to store group link for all members, rather than having it separately.
+
+- On updating group link data with one relay link at a time vs waiting for all links.
+
+  - Overhead is minimal - one request to owner's SMP server per relay.
+  - Waiting for a relay to send relay link can take indefinitely long.
+  - In proposed protocol owner doesn't have to wait for links from all relays for simplicity and to minimize wait time - it allows owner to conclude group creation potentially earlier, in case some relays are stuck or offline (owner can add their links later, once they successfully send it).
+  - Alternatively owner could collect all relay links first and make a single update of group link data, then notify all relays at once.
+
+- Owner should reject contact requests to their group link.
```

Comment:
Server would reject it, we have a mechanism to lock it from accepting connection, possibly has some implementation gaps. Owner has to reject it too, but it's an additional check, it shouldn't ever receive requests.